Add support for publishing large packages to PackagePublisher

0xjunha opened the pull request:
### Description
<!-- Please describe your change and its motivation. -->

Note: The integration test will fail on localnet due to a dependency on [this PR](https://github.com/aptos-labs/aptos-core/pull/14930), which updates the function argument type in the large_packages.move module.

Note: Since the PackagePublisher doesn't include a Move compilation feature, users may face inconvenience having to manually derive object addresses where the large package will be deployed, as shown in the `examples/large_package_publisher.py` example.

- Updated `large_packages` Move module address
- Downsized chunk size to 60KB
- Made methods in PackagePublisher reusable in examples
- Added object deployment and upgrade options for chunked publishing
- Updated `large_package_publisher` example

### Test Plan
<!-- Please provide us with clear details for verifying that your changes work. -->
```
make integration_test
```
### Related Links
<!-- Please link to any relevant issues or pull requests! -->

davidiw reviewed (commented):
this is still draft, let me know when it is ready for review.

0xjunha commented:
Dependency resolved, and now is ready for review! @davidiw 
Some more updates:
- Refactored the `PackagePublisher` class:
  - Introduced two helper classes:
    - `PublishHelper`
    - `CompileHelper`
- Added `compile_and_publish_move_package` method:
  - This encapsulates the process of compiling and publishing Move packages. It supports: 
    - Deployment to Account or Object
    - Regular or chunked publishing strategies
- Updated `large_package_publishing` example to utilize the `compile_and_publish_move_package`.
